lib/bcdice/common_command/add_dice/node.rb

```diff
@@ -53,21 +53,28 @@ def eval(game_system, randomizer)
             randomizer = Randomizer.new(randomizer, game_system)
             total = @lhs.eval(randomizer)
 
-            output =
-              if randomizer.rand_results.size <= 1 && @lhs.is_a?(Node::DiceRoll)
-                "(#{self}) ＞ #{total}"
-              else
-                "(#{self}) ＞ #{@lhs.output} ＞ #{total}"
+            interrim_expr =
+              unless randomizer.rand_results.size <= 1 && @lhs.is_a?(Node::DiceRoll)
+                @lhs.output
               end
 
-            if @cmp_op
-              rhs = @rhs.eval(nil)
-              output += game_system.check_result(total, randomizer.rand_results, @cmp_op, rhs)
-            end
+            result =
+              if @cmp_op
+                rhs = @rhs.eval(nil)
+                game_system.check_result(total, randomizer.rand_results, @cmp_op, rhs)
+              end
+            result ||= Result.new
```

Comment:
比較演算子がない場合は再代入ではなく `else` 節を使うことで、一度に代入できそうです。

```suggestion
            result =
              if @cmp_op
                rhs = @rhs.eval(nil)
                game_system.check_result(total, randomizer.rand_results, @cmp_op, rhs)
              else
                Result.new
              end
```

Reply:
`game_system.check_result` が `nil` を返すことがあるので再代入する処理にしていました。 ``game_system.check_result` は処理されたのか区別できるようにしておきたいので、このままにさせてください。